### ISSUE TITLE
merge=union

### ISSUE THREAD

**Issue by jamescostian** [olivierlacan/keep-a-changelog#56]
I read [this blog post](https://about.gitlab.com/2015/02/10/gitlab-reduced-merge-conflicts-by-90-percent-with-changelog-placeholders/) which talked about how keeping a CHANGELOG can prevent automatically accepting pull-requests, unless you put something like `CHANGELOG.md merge=union` in your `.gitattributes` file. Since this website is all about CHANGELOGs, I think it's worth mentioning somewhere in the site that specifying `CHANGELOG.md merge=union` is pretty important :smile:

**Comment by eosrei** [comment]
Works great on local git, but Github shows merge conflicts. Apparently github's merge tools don't support this. I hope I am wrong...

**Comment by silenceisgolden** [comment]
Stumbled on this a few days ago. It seems like GitHub does not allow this but GitLab might work better (TL:DR).

**Comment by glensc** [comment]
yes, in gitlab it works now: https://github.com/isaacs/github/issues/487#issuecomment-324538695

**Comment by silenceisgolden** [comment]
Is there any update for a workaround with GitHub? Even with the .gitattributes fix we are seeing a huge velocity hit by having to update every PR after a Changelog update has been merged.

From GitHub support

    Hi AJ,

    Thanks for reaching out to GitHub Support!

    I'm afraid what you're seeing is currently expected behavior.

    GitHub doesn't consider user-defined .gitattributes files (normally, we use our own .gitattrbutes file which you can't change). I agree it would be useful if you could define that yourself and I'll pass your feedback to the team, but I can't promise when such a feature might be available.

    The only workaround I could suggest is to merge pull requests in your local clone (and not via the web UI), by using the instructions shown in the merge box for each pull request. It's not a very elegant solution, I agree, but something to consider if this is causing you problems.

    Thanks for the feedback here, we really appreciate it. Please feel free to let us know if there is anything else we can help with!

**Comment by moreati** [comment]
As a live test of whether Github UI added support `merge=union` I created https://github.com/moreati/changelog-merge-test/pull/1, and https://github.com/moreati/changelog-merge-test/pull/2. Based on that I believe (as of 7 Feb 2021) the answer is still no.
